### DISCUSSION TITLE
How to have a snapshot with resolved reference

### DISCUSSION THREAD

**Discussion by Ziltosh** [mobxjs/mobx-state-tree#1935] [Q&A] [answered]
Hello,

I need to get a snapshot of a model to send to a backend, I use getSnapshot that works great but it not resolve references.
How can I have a "full" object representation of my current model without recreate it "by hand" ?

Thank you.

**Comment by EmilTholin** [comment]
Hi @Ziltosh!

Could you elaborate a bit on what you mean by "resolved references", and maybe give an example? I don't quite follow.

Do you mean that the snapshot should look more like `{ myRef: { id: "1", name: "foo", ... } }` rather than `{ myRef: "1" }`?

**Comment by Ziltosh** [comment]
Hi,

Yes, I would like to be able to recover the entire object and not only reference id, but I don't know how.

Here is a simple example: https://codesandbox.io/s/clever-currying-9qi28d?file=/src/index.ts

**Comment by k-ode** [comment, marked as answer]
You can use the `toJS` utility method from mobx: https://mobx.js.org/api.html#tojs

Note that it won't work in your current example becuase the reference needs to be resolvable from the same tree.

**Reply by EmilTholin** [reply]
Awesome! [Example](https://codesandbox.io/s/awesome-northcutt-g34kj2?file=/src/index.ts)

**Comment by Ziltosh** [comment]
Oh, thank you 👍  .
